Assets/HoloToolkit/Input/Scripts/Utilities/Interactions/TwoHandManipulatable.cs

```diff
@@ -0,0 +1,454 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using HoloToolkit.Unity.InputModule;
+using HoloToolkit.Unity.UX;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+
+namespace HoloToolkit.Unity.InputModule.Utilities.Interations
+{
+    /// <summary>
+    /// This script allows for an object to be movable, scalable, and rotatable with one or two hands. 
+    /// You may also configure the script on only enable certain manipulations. The script works with 
+    /// both HoloLens' gesture input and immersive headset's motion controller input.
+    /// See Assets/HoloToolkit-Examples/Input/Readme/README_TwoHandManipulationTest.md
+    /// for instructions on how to use the script.
+    /// </summary>
+    public class TwoHandManipulatable : MonoBehaviour, IInputHandler, ISourceStateHandler
+    {
+        /// <summary>
+        /// Event that gets raised when user begins manipulating the object
+        /// </summary>
+        public event Action StartedManipulating;
+
+        /// <summary>
+        /// Event that gets raised when the user ends manipulation
+        /// </summary>
+        public event Action StoppedManipulating;
+
+        [SerializeField]
+        [Tooltip("Transform that will be dragged. Defaults to the object of the component.")]
+        private Transform HostTransform = null;
+
+        [SerializeField]
+        [Tooltip("To visualize the object bounding box, drop the HoloToolKit/UX/Prefabs/BoundingBoxes/BoundingBoxBasic.prefab here. This is optional.")]
+        private BoundingBox boundingBoxPrefab = null;
+
+        /// <summary>
+        /// Reference to the Prefab from which clone is instantiated.
+        /// </summary>
+        public BoundingBox BoundingBoxPrefab
+        {
+            set
+            {
+                boundingBoxPrefab = value;
+            }
+
+            get
+            {
+                return boundingBoxPrefab;
+            }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public enum TwoHandedManipulation
```

Comment:
per coding guidelines, let's either make this private or move it into it's own class.